src/main/java/org/stellar/sdk/LiquidityPoolDepositOperation.java

```diff
@@ -0,0 +1,212 @@
+import org.stellar.sdk.xdr.OperationType;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Represents <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html#liquidity-pool-deposit" target="_blank">LiquidityPoolDeposit</a> operation.
+ * @see <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html" target="_blank">List of Operations</a>
+ */
+public class LiquidityPoolDepositOperation extends Operation {
+    private final LiquidityPoolID liquidityPoolID;
+    private final String maxAmountA;
+    private final String maxAmountB;
+    private final Price minPrice;
+    private final Price maxPrice;
+
+    private LiquidityPoolDepositOperation(LiquidityPoolID liquidityPoolID, String maxAmountA, String maxAmountB, Price minPrice, Price maxPrice) {
+      this.liquidityPoolID = checkNotNull(liquidityPoolID, "liquidityPoolID cannot be null");
+      this.maxAmountA = checkNotNull(maxAmountA, "amountA cannot be null");
+      this.maxAmountB = checkNotNull(maxAmountB, "amountB cannot be null");
+      this.minPrice = checkNotNull(minPrice, "minPrice cannot be null");
+      this.maxPrice = checkNotNull(maxPrice, "maxPrice cannot be null");
+    }
+
+    public LiquidityPoolID getLiquidityPoolID() {
+        return liquidityPoolID;
+    }
+
+    public String getMaxAmountA() {
+        return maxAmountA;
+    }
+
+    public String getMaxAmountB() {
+        return maxAmountB;
+    }
+
+    public Price getMinPrice() {
+        return minPrice;
+    }
+
+    public Price getMaxPrice() {
+        return maxPrice;
+    }
+
+    @Override
+    OperationBody toOperationBody(AccountConverter accountConverter) {
+        LiquidityPoolDepositOp op = new LiquidityPoolDepositOp();
+        op.setLiquidityPoolID(this.getLiquidityPoolID().toXdr());
+        op.setMaxAmountA(new org.stellar.sdk.xdr.Int64(Operation.toXdrAmount(this.getMaxAmountA())));
+        op.setMaxAmountB(new org.stellar.sdk.xdr.Int64(Operation.toXdrAmount(this.getMaxAmountB())));
+        op.setMinPrice(this.getMinPrice().toXdr());
+        op.setMaxPrice(this.getMaxPrice().toXdr());
+
+        OperationBody body = new org.stellar.sdk.xdr.Operation.OperationBody();
+        body.setDiscriminant(OperationType.LIQUIDITY_POOL_DEPOSIT);
+        body.setLiquidityPoolDepositOp(op);
+        return body;
+    }
+
+    /**
+     * Builds LiquidityPoolDeposit operation.
+     * @see LiquidityPoolDepositOperation
+     */
+    public static class Builder {
+        private Optional<LiquidityPoolID> liquidityPoolID = Optional.absent();
+        private Optional<Asset> a = Optional.absent();
+        private Optional<Asset> b = Optional.absent();
+        private String maxAmountA;
+        private String maxAmountB;
+        private Price minPrice;
+        private Price maxPrice;
+
+        private String mSourceAccount;
+
+        Builder(LiquidityPoolDepositOp op) {
```

Comment:
it seems like all these builder constructors should actually be constructors for LiquidityPoolDepositOperation. Each of these constructors has enough information from the parameters to construct a LiquidityPoolDepositOperation instance. Maybe we can move the constructors to `LiquidityPoolDepositOperation` and get rid of the builder entirely?

Reply:
I guess, it is better to be simple or consistent? Don't know what the java norm is.

Reply:
Usually builder constructors either don't have any parameters or have a subset of the full parameter set and allow you to provide the missing parameters via setters. In this case, I don't think we get any benefit from the builder because we require all the parameters up front